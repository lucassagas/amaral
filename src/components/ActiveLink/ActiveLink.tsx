import Link, { LinkProps } from 'next/link'
import { useRouter } from 'next/router'
import { cloneElement, ReactElement } from 'react'

type ActiveLinkProps = LinkProps & {
  children: ReactElement
  shouldMatchExactHref?: boolean
}

export function ActiveLink({
  children,
  shouldMatchExactHref = false,
  ...rest
}: ActiveLinkProps) {
  const router = useRouter()

  let isActive = false

  if (shouldMatchExactHref && router.asPath === rest.href) {
    isActive = true
  }

  return (
    <Link {...rest}>
      {cloneElement(children, {
        color: isActive ? 'orange.500' : 'gray.50',
      })}
    </Link>
  )
}
