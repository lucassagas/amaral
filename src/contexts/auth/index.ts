export { AuthContext, AuthContextProvider, handleSignOut } from './auth'
