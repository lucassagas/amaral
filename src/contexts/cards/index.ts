export { CardContext, CardContextProvider } from './cards';

