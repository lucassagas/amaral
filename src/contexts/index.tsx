import { theme } from '@/styles/theme'
import { ChakraProvider } from '@chakra-ui/react'
import { ReactNode } from 'react'
import { QueryClientProvider } from 'react-query'
import { ReactQueryDevtools } from 'react-query/devtools'
import { queryClient } from 'src/lib/queryClient'
import { AuthContextProvider } from './auth'

type AppProviderProps = {
  children: ReactNode
}

export function AppProvider({ children }: AppProviderProps) {
  return (
    <ChakraProvider theme={theme}>
      <AuthContextProvider>
        <QueryClientProvider client={queryClient}>
          {children}
          <ReactQueryDevtools />
        </QueryClientProvider>
      </AuthContextProvider>
    </ChakraProvider>
  )
}
