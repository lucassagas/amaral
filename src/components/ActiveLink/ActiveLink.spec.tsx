import { AppProvider } from '@/contexts/index'
import { Text } from '@chakra-ui/react'
import { render, screen } from '@testing-library/react'
import { ActiveLink } from './ActiveLink'

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '',
      pathname: 'example.com',
      query: '',
      asPath: 'example.com',
    }
  },
}))

describe('<ActiveLink />', () => {
  it('should be render correctly', () => {
    const { container } = render(
      <AppProvider>
        <ActiveLink href="example.com.br">
          <Text>Link</Text>
        </ActiveLink>
      </AppProvider>
    )

    expect(container).toMatchSnapshot()
  })

  it('should be render active link', () => {
    render(
      <AppProvider>
        <ActiveLink shouldMatchExactHref href="example.com" as="example.com">
          <p>Link</p>
        </ActiveLink>
      </AppProvider>
    )

    expect(screen.getByText(/link/i)).toHaveAttribute('color', 'orange.500')
  })

  it('should be render active link with shouldMatchExactHref false', () => {
    render(
      <AppProvider>
        <ActiveLink
          shouldMatchExactHref={false}
          href="example.com"
          as="example.com"
        >
          <p>Link</p>
        </ActiveLink>
      </AppProvider>
    )

    expect(screen.getByText(/link/i)).toHaveAttribute('color', 'orange.500')
  })
})
