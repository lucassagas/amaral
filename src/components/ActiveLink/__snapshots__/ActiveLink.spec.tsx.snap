// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActiveLink /> should be render correctly 1`] = `
<div>
  <p
    class="chakra-text css-53blzj"
  >
    Link
  </p>
</div>
`;
